contexts/custom-query-client-provider.tsx

```diff
@@ -0,0 +1,25 @@
+'use client'
+
+import getQueryClient from "@/components/customs/getQueryClient"
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+
+
+function CustomQueryClientProvider({ children }: { children: React.ReactNode }) {
+    // Why cant we use getQueryClient here?
+    const client = new QueryClient({
```

Comment:
we have a getQueryClient utility function. However I noticed in all our contexts we use this other query client that we generate each time --> This is not the best I think we may want a hook that creates and supplies a query client to our application.

I assume the issue with getQueryClient is that it lives on the server and is the query client for the server side data fetching. We need to init a new one on the client side so our client also has a query client and then use that (useQueryClient but it has the same functionality as getQueryClient?)

Sometime this server vs client stuff really throws me for a loop because there is not a very super clear distinction other than certain files being tagged as 'use_client' ... makes me wonder